lib/lev/errors.rb

```diff
@@ -4,12 +4,29 @@ module Lev
   #
   class Errors < Array
 
+    def initialize(routine_status = nil, raise_fatal_errors = false)
+      @routine_status = routine_status || BlackHoleStatus.new
+      @raise_fatal_errors = raise_fatal_errors
+    end
+
     def add(fail, args={})
       args[:kind] ||= :lev
       error = Error.new(args)
+
       return if ignored_error_procs.any?{|proc| proc.call(error)}
       self.push(error)
-      throw :fatal_errors_encountered if fail
+
+      routine_status.add_error(error, is_fatal: fail)
```

Comment:
if you want you can replace `add_error` with `add_fatal_error` and `add_nonfatal_error` instead.